content/en/blog/2022/merbridge/index.md

```diff
@@ -0,0 +1,163 @@
+
+In the past two years, eBPF becomes a hot-trending technology, and many projects based on eBPF have been raised to the community. Projects like Cilium, [px.dev](http://px.dev) provides us with great cases of eBPF in observability and network packet processing. With eBPF’s `sockops` and `redir` capabilities, data packets can be processed efficiently by directly being transported from the inbound socket to the outbound socket. In Istio scenarios, it will be possible to use eBPF to replace iptables rules to accelerate the data-plane traffic by shorten the data-path.
+
+Now that we have open sourced the Merbridge project, and by applying the following command to your Istio-managed cluster, you will directly get the eBPF ability to achieve network acceleration.
+
+{{< text bash >}}
+$ kubectl apply -f https://raw.githubusercontent.com/merbridge/merbridge/main/deploy/all-in-one.yaml
+{{< /text >}}
+
+{{< warning >}}
+Attention：Currently only support kernel version ≥ 5.7. Please check your kernel version to make sure eBPF functions properly.
+{{< /warning >}}
+
+### Utilize eBPF `sockops` for performance optimization
+
+Network connection is essentially socket communication. eBPF provides us with a function `bpf_msg_redirect_hash`, to directly forward the packets sent by the application in the inbound socket to the outbound socket, which can noticeably optimize the processing of the packets in the kernel.
+
+Here the `sock_map` is the crucial part to record the socket rules because an existing socket connection needs to be selected from the `sock_map` according to the current data packet information. Therefore, the socket information needs to be stored to the map at the hook of `sockops` or somewhere else, and being provided according to the key formed by given information(usually quadruple).
+
+## Approaches
+
+Next, we will introduce the detailed design and implementation principles of Merbridge step by step according to the actual scene, to give you a preliminary understanding of Merbridge and eBPF.
+
+### Istio sidecar traffic interception based on iptables
+
+{{< image link="./1.png" caption="Istio Sidecar Traffic Interception Based on Iptables" >}}
+
+As shown above, when external traffic accesses certain application’s ports, it will be intercepted by PREROUTING rule in iptables, forwarded to port 15001 of the sidecar container, and handed over to Envoy for processing(Like the red path of 1, 2, 3, 4 in the graph above).
```

Comment:
```suggestion
As shown above, when external traffic accesses certain application’s ports, it will be intercepted by PREROUTING rule in iptables, forwarded to port 15006 of the sidecar container, and handed over to Envoy for processing(Like the red path of 1, 2, 3, 4 in the graph above).
```